src/components/inputWithAutocomplete.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+
+class InputWithAutocomplete extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isVisibleHints: false
+        }
+
+        this.handleChange = this.handleChange.bind(this);
+        this.fillInputWithCheckedHint = this.fillInputWithCheckedHint.bind(this);
+        this.closeHints = this.closeHints.bind(this);
+        this.findHints = this.findHints.bind(this);
+    }
+
+    handleChange(event) {
+        let inputValue = event.target.value;
+
+        this.setState({
+            isVisibleHints: this.findHints(inputValue).length !== 0
```

Comment:
So you perform a walkthrough on options array to just determine whether there are hints or not? That's a resourceful operation, considering the fact that you repeat it again in render method